File: ios/ImagesHandling/ImagesHandling/Glasses/ImagesHandlingScreen.swift
//Copyright (c) 2020 Everysight LTD

import Foundation
import EvsKit
import NativeEvsKit


class ImagesHandlingScreen:Screen {

    override func onCreate() {
        let banana = Image()
            .setResource(image: ImgSrc(nameWithExtension: "banana.jpg", slot: .s1))
            .setX(x: 70.0)
            .setY(y: 130.0)
        add(uiElement: banana)

        let apple = Image()
            .setResource(image: ImgSrc(nameWithExtension: "apple.jpg", slot: .s2))
            .setX(x: 200.0)
            .setY(y: 70.0)
        add(uiElement: apple)
    }
}

File: android/ImagesHandling/src/main/java/com/everysight/sdk/samples/imageshandling/ImagesHandlingScreen.kt
package com.everysight.sdk.samples.imageshandling

import UIKit.app.Screen
import UIKit.app.data.*
import UIKit.app.resources.ImgSrc
import UIKit.widgets.Image

class ImagesHandlingScreen:Screen() {

    override fun onCreate() {
        val banana = Image()
            .setResource(ImgSrc("banana.jpg",ImgSrc.Slot.s1))
            .setX(100f).setY(130f)
        add(banana)

        val appleSrc = ImgSrc("apple.jpg",ImgSrc.Slot.s2)

        val apple1 = Image().setResource(appleSrc).setX(200f).setY(110f)
        add(apple1)

        // Since apple1 and apple2 uses the same image src 'apple.jpg', we should use the same image source ('appleSrc')
        // so it will be uploaded to the glasses just once (same slot)
        val apple2 = Image()
            .setResource(appleSrc)
            .setX(200f).setY(150f)
        add(apple2)

    }
}